src/Components/Web.Extensions/src/InputFile/RemoteBrowserFileStream.cs

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.Web.Extensions
+{
+    internal class RemoteBrowserFileStream : BrowserFileStream
+    {
+        private readonly IJSRuntime _jsRuntime;
+        private readonly ElementReference _inputFileElement;
+        private readonly int _maxSegmentSize;
+        private readonly PipeReader _pipeReader;
+        private readonly CancellationTokenSource _fillBufferCts;
+        private readonly TimeSpan _segmentFetchTimeout;
+
+        private bool _isReadingCompleted;
+        private bool _isDisposed;
+
+        public RemoteBrowserFileStream(
+            IJSRuntime jsRuntime,
+            ElementReference inputFileElement,
+            BrowserFile file,
+            RemoteBrowserFileStreamOptions options,
+            CancellationToken cancellationToken)
+            : base(file)
+        {
+            _jsRuntime = jsRuntime;
+            _inputFileElement = inputFileElement;
+            _maxSegmentSize = options.SegmentSize;
+            _segmentFetchTimeout = options.SegmentFetchTimeout;
+
+            var pipe = new Pipe(new PipeOptions(pauseWriterThreshold: options.MaxBufferSize, resumeWriterThreshold: options.MaxBufferSize));
+            _pipeReader = pipe.Reader;
+            _fillBufferCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
```

Comment:
Do we need this wrapper? Isn't it enough to store the CancellationToken itself?

Reply:
I added this so that if the stream was disposed, we could cancel the segment fetching task